storage/db-tool/src/replay_on_archive.rs

```diff
@@ -0,0 +1,251 @@
+pub struct Opt {
+    #[clap(
+        long,
+        help = "The first transaction version required to be replayed and verified"
+    )]
+    start_version: Version,
+    #[clap(
+        long,
+        help = "The last transaction version required to be replayed and verified"
+    )]
+    end_version: Version,
+    #[clap(flatten)]
+    replay_concurrency_level: ReplayConcurrencyLevelOpt,
+    #[clap(long = "target-db-dir", value_parser)]
+    pub db_dir: PathBuf,
+    #[clap(flatten)]
+    pub rocksdb_opt: RocksdbOpt,
+    #[clap(long, default_value="2000", help = "The number of transactions to be replayed in a chunk")]
+    pub chunk_size: usize,
+}
+
+impl Opt {
+    pub async fn run(self) -> Result<()> {
+        AptosVM::set_concurrency_level_once(self.replay_concurrency_level.get());
+        let replay_start: Instant = Instant::now();
+        let aptos_db = AptosDB::open(
+            StorageDirPaths::from_path(self.db_dir.as_path()),
+            true,
+            NO_OP_STORAGE_PRUNER_CONFIG,
+            self.rocksdb_opt.into(),
+            false,
+            BUFFERED_STATE_TARGET_ITEMS,
+            DEFAULT_MAX_NUM_NODES_PER_LRU_CACHE_SHARD,
+            None,
+        )?;
+
+        let backup_handler = aptos_db.get_backup_handler();
+        let arc_db = Arc::new(aptos_db) as Arc<dyn DbReader>;
+        let (start, limit) =
+            Self::get_start_and_limit(&arc_db, self.start_version, self.end_version)?;
+        info!(
+            start_version = start,
+            end_version = start + limit,
+            "Replaying transactions."
+        );
+        let mut failed_txns = Vec::new();
+        let txn_iter = backup_handler.get_transaction_iter(start, limit as usize)?;
+        let mut cur_txns = Vec::new();
+        let mut expected_events = Vec::new();
+        let mut expected_writesets = Vec::new();
+        let mut expected_txn_infos = Vec::new();
+        let mut chunk_start_version = start;
+        for (idx, item) in txn_iter.enumerate() {
+            let (input_txn, expected_txn_info, expected_event, expected_writeset) = item?;
+            let is_epoch_ending = ParsedTransactionOutput::parse_reconfig_events(&expected_event)
+                .next()
+                .is_some();
+            cur_txns.push(input_txn);
+            expected_txn_infos.push(expected_txn_info);
+            expected_events.push(expected_event);
+            expected_writesets.push(expected_writeset);
+            if is_epoch_ending || cur_txns.len() >= self.chunk_size {
+                // execute the transactions
+                let executed_outputs = if chunk_start_version == 0 {
+                    AptosVM::execute_block_no_limit(
+                        cur_txns
+                            .iter()
+                            .map(|txn| SignatureVerifiedTransaction::from(txn.clone()))
+                            .collect::<Vec<_>>()
+                            .as_slice(),
+                        &EmptyStateView {},
```

Comment:
`.state_view_at_version(None)` is fine? -- I mean, it can be
`let state_view = arc_db.state_view_at_version(chunk_start_version.checked_sub(1))`